crates/ironrdp-dvc-pipe-proxy/src/windows/semaphore.rs

```diff
@@ -0,0 +1,80 @@
+use std::sync::Arc;
+
+use windows::Win32::Foundation::HANDLE;
+use windows::Win32::System::Threading::{CreateSemaphoreW, ReleaseSemaphore};
+
+use crate::windows::{Handle, WindowsError};
+
+/// RAII wrapper for WinAPI semaphore handle.
+#[derive(Debug, Clone)]
+pub(crate) struct Semaphore {
+    handle: Arc<Handle>,
+}
+
+/// SAFETY: It is safe to send semaphore HANDLE between threads.
+unsafe impl Send for Semaphore {}
```

Comment:
https://github.com/Devolutions/sspi-rs/pull/430#pullrequestreview-2850078587

> - Explain why the code is "sound", detailing the invariants and preconditions that are maintained.
>
> - Avoid assumptions without justifications.
>	- Do not use phrases like "we assume this is safe because…" or "this should be safe".
>	- Provide concrete evidence or reasoning that justifies the safety of the operation.